src/main/receiver.cc

```diff
@@ -1,116 +1,42 @@
-#include "../lib/receiverWithPipe.h"
-#include "../lib/receiverWithMq.h"
-#include <iostream>
-#include <string>
-#include <fcntl.h>
-#include <sys/stat.h>
-#include <sys/types.h>
-#include <unistd.h>
-#include <cstdlib>
-#include <fstream>
-#include <getopt.h>
-#include <cstdio>
-
-enum ipcTypes
-{
-    Pipe,
-    Queue,
-    SharedMemory,
-    OtherSituation
-};
-int dealWithArg(int argc, char **argv, std::string &return_fileName, ipcTypes &return_method);
+#include "src/lib/receiverWithPipe.h"
+#include "src/lib/receiverWithMsgq.h"
+#include "src/lib/receiverWithShmem.h"
+#include "src/lib/argsParser.h"
 
 int main(int argc, char **argv)
 {
-    std::string fileName;
-    ipcTypes method;
-    int resultOfDealWithArg = dealWithArg(argc, argv, fileName, method);
-    if (resultOfDealWithArg == -1)
-    {
-        std::cout << "dealWithArg error" << std::endl;
-        return -1;
-    }
+    ArgsParser argsParser(argc,argv);
+    std::string fileName = argsParser.getFileName();
+    ipcType method = argsParser.getMethod();
+    std::cout << "File Used To Receive:" << fileName << std::endl;
     switch (method)
     {
     case Pipe:
     {
+        std::cout << "Pipe Is Used:" << std::endl;
         ReceiverWithPipe receiver;
         receiver.receiveFile(fileName);
         break;
     }
     case SharedMemory:
     {
-        std::cout << "SharedMemory has not been implemented yet" << std::endl;
+        std::cout << "Shared Memory Is Used:" << std::endl;
+        ReceiverWithShmem receiver;
+        receiver.receiveFile(fileName);
         break;
     }
     case Queue:
     {
-//        ReceiverWithMq receiver;
-//        receiver.receiveFile(fileName);
+        std::cout << "Message Queue Is Used:" << std::endl;
+        ReceiverWithMsgq receiver;
+        receiver.receiveFile(fileName);
         break;
     }
     case OtherSituation:
-        break;
+        return -1;
     default:
         return -1;
     }
 
     return 0;
 }
```

Comment:
You should try/catch inside your main to take care about throws and be sure that destructors are called.